eulerpi/examples/heat/heat.py

```diff
@@ -0,0 +1,210 @@
+from typing import Optional
+
+import jax.numpy as jnp
+import numpy as np
+
+from eulerpi.core.model import ArtificialModelInterface, JaxModel
+
+
+class Heat(JaxModel):
+    """A two-dimensional anisotropic heat conduction equation model on a square domain with four dirichlet boundaries.
+    The model is defined by the following partial differential equation on the square spacial domain :math:`\\Omega = [0, 1]^2` on the time interval :math:`[0, 0.1]`:
+    .. math::
+        \\frac{\\partial u}{\\partial t} = \\div \\left( \\kappa \\nabla u \\right)
+    subject to
+    .. math::
+        u(x, y, t=0) = 0
+    and
+    .. math::
+        u(0, y, t) = 1, \\quad u(1, y, t) = 0, \\quad u(x, 0, t) = 1, \\quad u(x, 1, t) = 0
+    with the thermal conductivity matrix :math:`\\kappa` and the temperature :math:`u`.
+    Inference is performed on the entries of :math:`\\kappa`: param[0] = :math:`\\kappa_{11}`, param[1] = :math:`\\kappa_{22}`, param[2] = :math:`\\kappa_{12}`.
+    Spatial discretization uses a finite difference scheme with a uniform grid, time stepping is done using the explicit Euler method.
+    """
+
+    param_dim = 3
+    data_dim = 4
+
+    CENTRAL_PARAM = np.array([0.5, 0.5, 0.5])
+    PARAM_LIMITS = np.array([[1.0, 2.0], [1.0, 2.0], [0.0, 1.0]])
+
+    def __init__(
+        self,
+        central_param: np.ndarray = CENTRAL_PARAM,
+        param_limits: np.ndarray = PARAM_LIMITS,
+        name: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        """Default constructor for the heat model.
+
+        Args:
+            central_param (np.ndarray, optional): Central parameter within the parameter domain, where the density is non-zero. Defaults to CENTRAL_PARAM.
+            param_limits (np.ndarray, optional): Maximum parameter limits for sampling or grid-based inference. The parameter domain is contained within the box given by param_limits. Defaults to PARAM_LIMITS.
+            name (Optional[str], optional): Name for the model. Defaults to None means the model name is "heat".
+        """
+        super().__init__(central_param, param_limits, name=name, **kwargs)
+
+    def get_param_bounds(self) -> np.ndarray:
+        """Getter for the parameter limits.
+
+        Returns:
+            np.ndarray: An array containing the lower and upper limits for each parameter.
+        """
+        return self.param_limits
+
+    @classmethod
+    def forward(self, param: np.ndarray) -> np.ndarray:
+        """Forward method for the heat model. Yields the solution of the anisotropic heat conduction equation at time :math:`\\t=0.1`
+        in five spacial points, which are arranged similar to the number "five" on a dice.
```

Comment:
I would go with the more common variant "spatial". Besides from that I am a bit confused here and would like to propose the some changes, if I am wrong pls correct.

- They points you mention are not really spatial points but points in the space-time.
- The number of evaluation points is 4, not five and you should probably define them next to the class variable, I think the number of evaluation points is the data_dim.
- Do you really evaluate it at t=0.1? The last index in your dice arrangement seems to be the time index according to the documentation of the function `perform_simulation`. Fix the documentation here too.



-

Reply:
Yes, good catch again, thanks. Looks like I've written the documentation beforehand and forgot to update the code accordingly :D The evaluation_points are now a class variable. I think the documentation of perform_simulation should be correct, as the full space-time grid is returned?

Reply:
yes again :+1: